### PR TITLE
ARROW-12138: [Go][IPC] Update flatbuffers definitions

### PR DESCRIPTION
**Pull request by zeroshade**
Updating the generated flatbuffer code so that newer features like compression in IPC can get implemented. Doing the updating of the flatbuffer generated code first as a separate change.

@sbinet @emkornfield

### CODE REVIEW

**Comment by zeroshade** [comment]
All that was done for this change was to run `go run ./gen-flatbuffers.go` followed by updating the couple of files in the ipc module which had to switch direct references to `byte` or `int16` to the exported types from the flatbuffers files. So the only files in this change that contain anything that wasn't generated by flatc are ipc/file_writer.go, ipc/message.go and ipc/metadata.go. Everything else just contains the generated flatbuffer code.

**Review by sbinet — APPROVED** [review]

**Comment by zeroshade** [comment]
@sbinet @emkornfield @wesm once this is merged, i have a PR ready for adding compression handling to the go IPC implementation, I'm just waiting for this to get merged before filing it for ease of review

**Pull request closed without merging**
